ntoskrnl/include/internal/i386/intrin_i.h

```diff
@@ -5,6 +5,45 @@ extern "C"
 {
 #endif
 
+FORCEINLINE
+PKGDTENTRY
+KiGetGdtEntry(
+    IN PVOID pGdt,
+    IN USHORT Selector)
+{
+    return (PKGDTENTRY)((ULONG_PTR)pGdt + (Selector & ~RPL_MASK));
+}
+
+FORCEINLINE
+VOID
+KiSetGdtDescriptorBase(
+    IN OUT PKGDTENTRY Entry,
+    IN UINT32 Base)
+{
+    Entry->BaseLow = (UINT16)(Base & 0xffff);
+    Entry->HighWord.Bytes.BaseMid = (UINT8)((Base >> 16) & 0xff);
+    Entry->HighWord.Bytes.BaseHi  = (UINT8)((Base >> 24) & 0xff);
```

Comment:
As you've certainly noticed, in the original code I had a commented `// Entry->BaseUpper = (ULONG)(Base >> 32);` : do you know/think whether this could be useful too?

Reply:
Any answer for my question?

Reply:
I just grabbed the ones from freeloader instead. I do not mind eitherway.

Reply:
oki :+1: